### ISSUE TITLE
The training recipe for SeqDiffuSeq without adaptive noise schedule

### ISSUE THREAD

**Issue by BaohaoLiao** [Yuanhy1997/SeqDiffuSeq#10]
Hi, I try to reproduce your result in Table 3, especially for SeqDiffuSeq without Adaptive Noise Schedule. Here is how I reproduce it:

1. I remove these two lines:
https://github.com/Yuanhy1997/SeqDiffuSeq/blob/ed56ca480002b89a26ebd16b0257e1aa880f80f1/src/modeling/diffusion/gaussian_diffusion.py#L345
https://github.com/Yuanhy1997/SeqDiffuSeq/blob/ed56ca480002b89a26ebd16b0257e1aa880f80f1/src/modeling/diffusion/gaussian_diffusion.py#L295
After removing the first line, the noise schedule would not be changed during training. And after removing the second line, the model will not load saved noise schedule and only use the pre-defined one, i.e. the one in the init function. 

2. I use your default training and sampling script (sqrt noise schedule with uniform timestep sampling). And run sampling with different checkpoints, e.g. the 280K, 400K checkpoint. 

However, my best result is about 18, which is quite different from your reported number 28.94. May I ask what I should change to get the reported number? BTW, I can reproduce the result of SeqDiffuSeq.

**Comment by Yuanhy1997** [comment]
I think your way of modifying the code still adjust the alpha-cumprod parameters. Removing that two lines only prevent updating of other parameter such as betas, alphas. Maybe you should removing the lines corresponding to the schedule in the training loss function.

**Comment by BaohaoLiao** [comment]
I'm confused. The loss history update function should only update the loss history and has nothing to do with other parameters. It also updates the noise schedule by the first line shown above. https://github.com/Yuanhy1997/SeqDiffuSeq/blob/ed56ca480002b89a26ebd16b0257e1aa880f80f1/src/modeling/diffusion/gaussian_diffusion.py#L297

I just tried to set ```SCHEDULE_UPDATE_STRIDE=1000000``` and run the experiment. In this way, the noise schedule should not be changed. Will let you know my result later.

**Comment by BaohaoLiao** [comment]
I use the above-mentioned recipe, i.e. setting SCHEDULE_UPDATE_STRIDE=1000000, and don't load any alpha-cumprod file. More specifically, I use a pseudo alpha-cumprod file as input and delete this line:
https://github.com/Yuanhy1997/SeqDiffuSeq/blob/ed56ca480002b89a26ebd16b0257e1aa880f80f1/src/modeling/diffusion/gaussian_diffusion.py#L295

I estimate the 280K checkpoint. But the result is 16.37. Could you tell me what would be the reason for such a big difference between my number and your reported number?

**Comment by Yuanhy1997** [comment]
I actually have no clue how you modify the code and where the problem is. I think it would be a good way to debug that maybe you print the schedule numbers when training and inference, rather than just looking at a pale number.

**Comment by BaohaoLiao** [comment]
I actually do output the alpha-cumprod during training and make sure it is not modified. And during inference, I also check whether the model uses the sqrt or the modified alpha-cumprod. All settings should fulfill my requirement. That's why I'm confused about your reported number.

**Comment by Yuanhy1997** [comment]
I really want to help while the information provided is limited and I don't know what else can I offer.

**Comment by BaohaoLiao** [comment]
Could you offer me the training script you used for SeqDiffuSeq without adaptive noise schedule? If the codebase for SeqDiffuSeq without adaptive noise schedule is different from current codebase, could you upload it as a new branch?

The thing I tried is basically deleting these two lines (Though I also tried something else,  they should all go in this direction.):
https://github.com/Yuanhy1997/SeqDiffuSeq/blob/ed56ca480002b89a26ebd16b0257e1aa880f80f1/src/modeling/diffusion/gaussian_diffusion.py#L345
https://github.com/Yuanhy1997/SeqDiffuSeq/blob/ed56ca480002b89a26ebd16b0257e1aa880f80f1/src/modeling/diffusion/gaussian_diffusion.py#L295
Then exactly follow your README.

About "I think your way of modifying the code still adjust the alpha-cumprod parameters. Removing that two lines only prevent updating of other parameter such as betas, alphas." It's not true. I print out the alpha-cumprod. And it is not updated during training after deleting the above two lines. 

About "Maybe you should removing the lines corresponding to the schedule in the training loss function." I also did this. It leads to the same result as deleting the above two lines.

**Comment by Yuanhy1997** [comment]
I will work on that. I cleaned the code a bit prior to releasing. The codebase for ablations is similar.

**Comment by BaohaoLiao** [comment]
Great! Look forward to it.

**Comment by chenxwh** [comment]
Hi @Yuanhy1997 I am wondering as well if the training code without the adaptive noise is available now?

**Comment by Yuanhy1997** [comment]
Still, it's pending for approval.
